pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ConsumerHandler.java

```diff
@@ -240,12 +242,27 @@ public void onWebSocketText(String message) {
                 // We should have received an ack
                 MessageId msgId = MessageId.fromByteArrayWithTopic(Base64.getDecoder().decode(command.messageId),
                         topic.toString());
-                consumer.acknowledgeAsync(msgId).thenAccept(consumer -> numMsgsAcked.increment());
-                if (!this.pullMode) {
-                    int pending = pendingMessages.getAndDecrement();
-                    if (pending >= maxPendingMessages) {
-                        // Resume delivery
-                        receiveMessage();
+                if ("cumulative".equals(command.ackType)) {
+                    ConsumerStats cStats = consumer.getStats();
+                    long ackDiff = cStats.getTotalMsgsReceived() - cStats.getTotalAcksSent();
```

Comment:
These are "statistics" and are reset to 0 periodically (the default interval is 60 seconds). So I don't think these values should be used in this case.
https://github.com/apache/pulsar/blob/988e03e07da4e31ff5b40ddc1647c7d64dec79f4/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerStatsRecorderImpl.java#L114-L159

Reply:
Can you recommend a different interface for getting the current offset based on a message ID? If I have that, then I can still calculate the difference for the number of messages acked by the consumer and update the value.

Reply:
Umm... I don't think there is such an interface.
We can find the number of messages between the two message IDs on the broker side, but it is difficult on the websocket proxy side.

Reply:
Is it infeasible to query the broker here to gather that information?

Reply:
The REST API provided by the broker contains the number of messages that have not yet been acknowledged. However, I don't think it's realistic in terms of performance to access the broker's REST API every time the WebSocket proxy receives a cumulative ack.
```sh
$ curl -s http://dev-broker01.pulsar.xxx.yahoo.co.jp:8080/admin/v2/persistent/massakam/test/t1/internalStats | jq .

{
  "entriesAddedCounter": 10,
  "numberOfEntries": 30,
  "totalSize": 1760,
  "currentLedgerEntries": 10,
  "currentLedgerSize": 520,
  "lastLedgerCreatedTimestamp": "2020-04-14T10:47:40.829+09:00",
  "waitingCursorsCount": 2,
  "pendingAddEntriesCount": 0,
  "lastConfirmedEntry": "2823032:9",
  "state": "LedgerOpened",
  "ledgers": [
    {
      "ledgerId": 2820274,
      "entries": 20,
      "size": 1240,
      "offloaded": false
    },
    {
      "ledgerId": 2823032,
      "entries": 0,
      "size": 0,
      "offloaded": false
    }
  ],
  "cursors": {
    "pulsar.repl.dev2": {
      "markDeletePosition": "2823032:9",
      "readPosition": "2823032:10",
      "waitingReadOp": true,
      "pendingReadOps": 0,
      "messagesConsumedCounter": 10,
      "cursorLedger": 2823043,
      "cursorLedgerLastEntry": 1,
      "individuallyDeletedMessages": "[]",
      "lastLedgerSwitchTimestamp": "2020-04-14T10:47:40.831+09:00",
      "state": "Open",
      "numberOfEntriesSinceFirstNotAckedMessage": 1,
      "totalNonContiguousDeletedMessagesRange": 0,
      "properties": {}
    },
    "sub1": {
      "markDeletePosition": "2820274:19",
      "readPosition": "2823032:10",
      "waitingReadOp": true,
      "pendingReadOps": 0,
      "messagesConsumedCounter": 0,
      "cursorLedger": 2823042,
      "cursorLedgerLastEntry": 0,
      "individuallyDeletedMessages": "[]",
      "lastLedgerSwitchTimestamp": "2020-04-14T10:52:11.384+09:00",
      "state": "Open",
      "numberOfEntriesSinceFirstNotAckedMessage": 11,
      "totalNonContiguousDeletedMessagesRange": 0,
      "properties": {}
    }
  }
}
```

Reply:
Understood. I’m not sure this PR makes much sense then, given that.